neqo-http3/tests/send_message.rs

```diff
@@ -0,0 +1,291 @@
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    send_headers(&mut request).unwrap();
+    request.send_data(RESPONSE_DATA).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    send_trailers(&mut request).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    request.close_send().unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    process_client_events(&mut hconn_c);
+}
+
+#[test]
+fn response_trailers_no_data() {
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    request
+        .send_headers(&[Header::new(":status", "200"), Header::new("something", "3")])
+        .unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    send_trailers(&mut request).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    request.close_send().unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    process_client_events_no_data(&mut hconn_c);
+}
+
+#[test]
+fn multiple_response_trailers() {
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    send_headers(&mut request).unwrap();
+    request.send_data(RESPONSE_DATA).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    send_trailers(&mut request).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+
+    assert_eq!(send_trailers(&mut request), Err(Error::InvalidInput));
+
+    request.close_send().unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    process_client_events(&mut hconn_c);
+}
+
+#[test]
+fn data_after_tralier() {
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    send_headers(&mut request).unwrap();
+    request.send_data(RESPONSE_DATA).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    send_trailers(&mut request).unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+
+    assert_eq!(request.send_data(RESPONSE_DATA), Err(Error::InvalidInput));
+
+    request.close_send().unwrap();
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    process_client_events(&mut hconn_c);
+}
+
+#[test]
+fn trailers_after_close() {
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    send_headers(&mut request).unwrap();
+    request.send_data(RESPONSE_DATA).unwrap();
+    request.close_send().unwrap();
+
+    assert_eq!(send_trailers(&mut request), Err(Error::InvalidStreamId));
+
+    exchange_packets(&mut hconn_c, &mut hconn_s);
+    process_client_events(&mut hconn_c);
+}
+
+#[test]
+fn multiple_response_headers() {
+    let (mut hconn_c, mut hconn_s, mut request) = connect_send_and_receive_request();
+    request
+        .send_headers(&[Header::new(":status", "200"), Header::new("something", "3")])
+        .unwrap();
+
+    assert_eq!(
+        request.send_headers(&[Header::new(":status", "200"), Header::new("something", "3"),]),
+        Err(Error::InvalidHeader)
+    );
```

Comment:
Can you add another test with an information status code after a final status code?